app/uk/gov/hmrc/economiccrimelevyregistration/controllers/EntityTypeController.scala

```diff
@@ -70,17 +71,45 @@ class EntityTypeController @Inject() (
             )
 
           eclRegistrationConnector
-            .upsertRegistration(
-              dataCleanup.cleanup(
-                request.registration.copy(
-                  entityType = Some(entityType)
-                )
-              )
-            )
+            .upsertRegistration(cleanup(mode, request.registration, entityType, previousEntityType))
             .flatMap { updatedRegistration =>
-              pageNavigator.nextPage(mode, updatedRegistration).map(Redirect)
+              previousEntityType match {
+                case Some(value) if value == entityType && EntityType.isOther(entityType) && mode == CheckMode =>
+                  pageNavigator.navigateToCheckYourAnswers().map(Redirect)
+                case _                                                                                         =>
+                  pageNavigator.nextPage(mode, updatedRegistration).map(Redirect)
+              }
             }
         }
       )
   }
+
+  private def cleanup(
+    mode: Mode,
+    registration: Registration,
+    entityType: EntityType,
+    previousEntityType: Option[EntityType]
+  ) = {
+    val isOther = EntityType.isOther(entityType)
```

Comment:
This is a bit hard to follow, you could use pattern matching instead:

```
    (previousEntityType, mode) match {
      case (Some(entityType), CheckMode) if EntityType.isOther(entityType)          => registration
      case (Some(entityType), CheckMode)                                            =>
        dataCleanup.cleanup(
          registration.copy(
            entityType = Some(entityType)
          )
        )
      case (Some(entityType), _) if previousEntityType.contains(entityType) =>
        dataCleanup.cleanup(
          registration.copy(entityType = Some(entityType))
        )
      case _                                                                        =>
        dataCleanup.cleanupOtherEntityData(
          registration.copy(entityType = Some(entityType))
        )
    }

```

Reply:
It looks like we're doing the same thing when previous entity type matches the entity type and when it doesn't so reading it it would seem that it would make most sense to have
 ```
case Some(Some(entityType),_) if EntityType.isOther(entityType) => 

          dataCleanup.cleanupOtherEntityData(
            registration.copy(entityType = Some(entityType))
          )
```

Reply:
I find pattern matching on several things at once more confusing.